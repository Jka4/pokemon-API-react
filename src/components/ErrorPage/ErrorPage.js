import React from "react";
import './styles/Style.scss'

const ErrorPage = () => {

	return (
		<div className='errorPage' >
			<div className="errCode">404</div>
			<div className="image">
				<img src="https://github.com/Jka4/pokemon-API-react/blob/master/public/images/404.png?raw=true" alt="" />
			</div>
			<div className="description">Page Not Found</div>

		</div>
	);
}

export default ErrorPage;
