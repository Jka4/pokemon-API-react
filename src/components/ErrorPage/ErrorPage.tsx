import React from 'react';

import styled from 'styled-components/macro';

const ErrorPage: React.FC = () => {
  return (
    <ErrorPageStyled>
      <div className="errCode">404</div>
      <div className="image">
        <Image src={`${process.env.PUBLIC_URL}/images/404.png`} alt="404_pickachu" />
      </div>
      <div className="description">Page Not Found</div>
    </ErrorPageStyled>
  );
};

const ErrorPageStyled = styled.div`
  width: 100%;
  min-height: 700px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  .errCode {
    font-size: 42px;
  }

  .description {
    font-size: 24px;
  }
`;

const Image = styled.img`
  filter: drop-shadow(13px 17px 45px #222);
`;

export default ErrorPage;
