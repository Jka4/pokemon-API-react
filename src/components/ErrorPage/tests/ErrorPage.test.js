import React from "react";
import ErrorPage from "@ErrorPage";
import renderer from "react-test-renderer";

describe("ErrorPage", () => {
  it("ErrorPage renders correctly", async () => {
    const tree = renderer.create(<ErrorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
