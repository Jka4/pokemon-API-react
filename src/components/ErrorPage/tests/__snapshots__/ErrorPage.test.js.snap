// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage ErrorPage renders correctly 1`] = `
<div
  className="errorPage"
>
  <div
    className="errCode"
  >
    404
  </div>
  <div
    className="image"
  >
    <img
      alt="404_pickachu"
      loading="lazy"
      src="/images/404.png"
    />
  </div>
  <div
    className="description"
  >
    Page Not Found
  </div>
</div>
`;
